lib/esm/connection/client.rb

```diff
@@ -58,14 +59,16 @@ def send_error(content, block: false)
       # @param block [true/false] Cause this method to block the current thread and either
       #   1. Until the request is responded to by the client
       #   2. The timeout is reached. This will raise ESM::Exception::RejectedPromise
+      # @param timeout [Integer] A number in seconds on how long the process will block before
+      #   considering a message to be timed out. Defaults to response_timeout in config.yml
       #
       # @return [ESM::Connection::Promise, ESM::Message]
       #   If block is false, a promise in an processing status is returned
       #   If block is true, the response as ESM::Message is returned
       #
       # @raises ESM::Exception::RejectedPromise, ESM::Exception::ExtensionError
       #
-      def send_request(message = nil, type:, block: true)
+      def send_request(message = nil, type:, block: true, timeout: @config.response_timeout)
         # I feel so dirty. Multiline unless statements *shudder*
```

Comment:
why is this bad? it's easily readable and does what it's supposed to, right?

Reply:
Nothing is wrong with the code, this is down to personal choice. I've never been a fan of `unless` statements but have slowly started to use the single line variants here and there because I do see their value. However, multi-line versions of them are not my preference so having one in my code-base that I added makes me feel gross lol.